api/grpcserver/globalstate_service.go

```diff
@@ -271,23 +271,45 @@ func (s GlobalStateService) AccountDataStream(in *pb.AccountDataStreamRequest, s
 
 	// Subscribe to the various streams
 	var (
-		channelAccount chan types.Address
-		channelReward  chan events.Reward
-		channelReceipt chan events.TxReceipt
+		accountCh       <-chan interface{}
+		rewardsCh       <-chan interface{}
+		receiptsCh      <-chan interface{}
+		accountBufFull  <-chan struct{}
+		rewardsBufFull  <-chan struct{}
+		receiptsBufFull <-chan struct{}
```

Comment:
why not to reuse same channel and return error on `!ok` ? if stream is terminated by the caller it will exit on `stream.Context().Done()`

Reply:
but why would it close the channel? i don't think that there is any logic to do that

Reply:
it shouldn't, I just thought it's safer not to rely on it

Reply:
but I agree that we can assume that it wouldn't close the channel and get rid of the second channel